### PR TITLE
Document the reducer and add more links to storybook examples

### PR DESCRIPTION
**Pull request by gregjopa**
This PR adds some documentation to the readme about the reducer hook. Preview the changes here: https://github.com/paypal/react-paypal-js/blob/document-reducer/README.md#paypalbuttons

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,69 @@ export default function App() {
...
+
+To learn more, check out the [loading spinner example in storybook](https://paypal.github.io/react-paypal-js/?path=/story/example-usepaypalscriptreducer--loading-spinner).
+
+#### Reloading when parameters change
+
+The `usePayPalScriptReducer` hook can be used to reload the JS SDK script when parameters like currency change. It provides the action `resetOptions` for reloading with new parameters. For example, here's how you can use it to change currency.
+
+```js
+const [{ scriptOptions }, dispatch] = usePayPalScriptReducer();
+const [value, setValue] = useState(scriptOptions.currency);
```
Comment (mnicpt) [inline README.md:100]: What about `const [currency, setCurrency]`?

File: README.md [hunk 2]
```diff
@@ -60,6 +60,69 @@ export default function App() {
...
+
+function onChange({ target: { value } }) {
+    setValue(value);
+    dispatch({
+        type: "resetOptions",
+        value: {
+            ...scriptProviderOptions,
+            currency: value,
```
Comment (mnicpt) [inline README.md:108]: If above, then just `currency` here.

File: README.md [hunk 3]
```diff
@@ -60,6 +60,69 @@ export default function App() {
...
+
+#### Reloading when parameters change
+
+The `usePayPalScriptReducer` hook can be used to reload the JS SDK script when parameters like currency change. It provides the action `resetOptions` for reloading with new parameters. For example, here's how you can use it to change currency.
+
+```js
+const [{ scriptOptions }, dispatch] = usePayPalScriptReducer();
+const [value, setValue] = useState(scriptOptions.currency);
+
+function onChange({ target: { value } }) {
```
Comment (mnicpt) [inline README.md:102]: `onCurrencyChange`?

File: README.md [hunk 4]
```diff
@@ -60,6 +60,69 @@ export default function App() {
...
+        value: {
+            ...scriptProviderOptions,
+            currency: value,
+        },
+    });
+}
+
+return (
+    <>
+        <select value={value} onChange={onChange}>
```
Comment (mnicpt) [inline README.md:115]: `onCurrencyChange`?

**Review by mnicpt — APPROVED** [review]
Left a few non-blocking comments.

**Comment by gregjopa** [comment]
Thanks for the suggestions @mnicpt! I made those changes and pushed them up. I agree that the new naming helps the code be more self-describing 😄

File: README.md (hunk 2 above)
Comment (gregjopa) [reply README.md:108]: This one is actually referring to the new `event.target.value` coming in from the change handler. So it should remain `value`

**Pull request merged**